src/xml_parser_main_metadata_together.c

```diff
@@ -52,30 +77,19 @@ call_user_callback_if_package_finished(cr_Package *pkg, cr_CbData *cb_data, GErr
             g_hash_table_remove(cb_data->in_progress_pkgs_hash, pkg->pkgId);
         } else {
             //remove first element in the list
-            cb_data->in_progress_pkgs_list = cb_data->in_progress_pkgs_list->next;
+            g_slist_delete_link(cb_data->in_progress_pkgs_list, cb_data->in_progress_pkgs_list);
```

Comment:
I think we have to also assign the new head of the list.
```suggestion
            cb_data->in_progress_pkgs_list = g_slist_delete_link(cb_data->in_progress_pkgs_list, cb_data->in_progress_pkgs_list);
```